juju/juju
# Consolidate exec actions

Make juju exec and juju run use the same APIs and CLI UX.
This means that juju exec will now have the new, improved actions behaviour.
The exec and actions implementation both used actions at the backed, but totally different CLI code, and also slightly different api results when enqueuing tasks.

There's a lot of code moving around which blows out the diff size. Plus github is showing the move of exec.go and exec_test.go as a add / delete instead of a move.

The key parts in separate commits:
- remove the state code to support action UUIDs
- the api facade changes to make enqueuing juju exec actions behave the same as enqueuing charm actions
- moving exec from commands to actions package and factoring out common code for results handing and tests

Also, the actions tests were using a real clock, which was problematic to start with, but more so when trying to get the tests right. The real clock in tests is replaced with a test clock, which resulted in the test helpers needing to run the commands in a separate goroutine so the test clock cloud be advanced.

## QA steps

bootstrap and deploy a charm with actions
check things like
```
juju exec --machine 0 ps
juju exec --machine 0 ps --wait 1s
juju run ubuntu-lite/0 hello
juju run ubuntu-lite/0 hello --wait 1s
```
Note that actions now will print the receivers which did not completed when a timeout occurs (as a result of consolidating the run and exec code).
